.github/workflows/benchmarks.yml

```diff
@@ -168,7 +168,7 @@ jobs:
     container:
       image: matterlabs/llvm_runner:ubuntu22-llvm17-latest
     needs: benchmarks
-    if: always()
+    if: failure() || success()

```

Comment:
Let me explain the logic:

* We're running `reference` and `candidate` benchmark jobs
* Some of the tests in each of them can fail, but the other tests will produce meaningful benchmark results
* We would like to report that the benchmark job failed with the red checks, but at the same time it makes sense to generate a comparison report for the last part of the tests we successfully executed and measured performance for
* This is exactly why we specify `failure()` - to catch this particular case.

but `always()` doesn't work, because canceling or aborting jobs will cause the comparison to run unconditionally, we don't want that.

Reply:
So does it change the behavior so that the benchmark report will be printed even if there are failures?

Reply:
If so, I do not object, but IIRC the tester doesn't emit the report if there are failures.

Reply:
So, yeah. The report will be printed for all tests that passed, because JSON will be generated.